### PR TITLE
Add secondary telemetry to InlineController

### PR DESCRIPTION
**Pull request by nathan-downs**
## Test plan

run extension in debug mode
check test eventlogs on dotcom for event emission
check bigquery for event reception

### CODE REVIEW

File: lib/shared/src/telemetry-v2/TelemetryRecorderProvider.ts [hunk 1]
```diff
@@ -135,3 +135,9 @@ class ConfigurationMetadataProcessor implements TelemetryProcessor {
         )
     }
 }
+
+export const NOOP_TELEMETRY_RECORDER_SERVICE: TelemetryRecorder = {
+    recordEvent() {
+        /* noop */
+    },
+}
```
Comment (bobheadxi) [inline lib/shared/src/telemetry-v2/TelemetryRecorderProvider.ts:143]: This already exists, a package-level `noOpTelemetryRecorder` is available

File: vscode/webviews/OnboardingExperiment.story.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { Meta, StoryObj } from '@storybook/react'
 
 import { NOOP_TELEMETRY_SERVICE } from '@sourcegraph/cody-shared/src/telemetry'
+import { NOOP_TELEMETRY_RECORDER_SERVICE } from '@sourcegraph/cody-shared/src/telemetry-v2/TelemetryRecorderProvider'
```
Comment (bobheadxi) [inline vscode/webviews/OnboardingExperiment.story.tsx:4]:
```suggestion
import { noOpTelemetryRecorder } from '@sourcegraph/cody-shared/src/telemetry-v2/TelemetryRecorderProvider'
```

File: vscode/webviews/utils/telemetry.ts [hunk 3]
```diff
@@ -12,3 +13,14 @@ export function createWebviewTelemetryService(vscodeAPI: VSCodeWrapper): Telemet
...
 }
+
+/**
+ * Create a new {@link TelemetryRecorder} for use in the VS Code webviews.
+ */
+export function createWebviewTelemetryRecorder(vscodeAPI: VSCodeWrapper): TelemetryRecorder {
+    return {
+        recordEvent: (eventName, properties) => {
+            vscodeAPI.postMessage({ command: 'event', eventName, properties: { properties } })
+        },
```
Comment (bobheadxi) [inline vscode/webviews/utils/telemetry.ts:24]: this doesn't correctly implement `TelemetryRecorder` right? 🤔 how does it work?

**Comment by dominiccooney** [comment]
Could you elaborate the test plan a bit? "check bigquery" and "check test event logs on dotcom" is a bit of a leap for people who aren't doing it regularly. I guess that means you're signed in to dotcom? Spell it out so people can reproduce your test plan.

Needless to say, it would be great if this had an automated test.

**Comment by dominiccooney** [comment]
It would be great to give a tiny bit of background about the motivation for this change, too, since it is not linked to an Issue. Yes, it adds another telemetry thing, but why are we doing that?

File: vscode/webviews/utils/telemetry.ts (hunk 3 above)
Comment (nathan-downs) [reply vscode/webviews/utils/telemetry.ts:24]: probably by calling the old version 😅

Comment (nathan-downs) [reply vscode/webviews/utils/telemetry.ts]:
```
/**
 * Create a new {@link TelemetryRecorder} for use in the VS Code webviews.
 */
export function createWebviewTelemetryRecorder(vscodeAPI: VSCodeWrapper): TelemetryRecorder {
    return {
        recordEvent: (feature, action, parameters?: TelemetryEventProperties) => {
            vscodeAPI.postMessage({
                command: 'telemetryEvent',
                feature,
                action,
                parameters,
            })
        },
    }
}
```

**Comment by nathan-downs** [comment]
@dominiccooney The general background of this effort is supporting implementation of the new [Telemetry framework](https://docs.sourcegraph.com/dev/background-information/telemetry) for the [#wg-event-logging-everywhere](https://sourcegraph.slack.com/archives/C05BGNBEPKL) project. We want to enable event telemetry by default, our old event logging mechanism wouldn't be able to support this, the Telemetry Gateway solves that issue and allows us to scale.

Adding the secondary telemetry recording allows us to ensure that we're not losing any data before we deprecate the old system.

We do have automated tests that check `loggedV2Events` for expected captured events. Example from `task-controller.test.ts`:
```
await expect.poll(() => loggedEvents).toEqual(expectedOrderedEvents)
await expect
    .poll(() => loggedV2Events)
    .toEqual([
        'cody.auth/failed',
        'cody.auth/connected',
        'cody.command.edit/executed',
        'cody.recipe.fixup/executed',
        'cody.fixup.apply/succeeded',
    ])
```
I can add more robust tests for all of the different reconfigured feature names. I have generally been running these locally with a config flag to enable the export of events I'm testing. Any event that is being tested will flow into the `event_logs` table then be exported to a pub/sub which feeds the data into the Telligentsourcegraph project (`telemetry.event_telemetry_raw` is the landing location).

**Comment by bobheadxi** [comment]
@dominiccooney some additional context:

- Background: https://docs.sourcegraph.com/dev/background-information/telemetry
- Introduction of new SDK: https://github.com/sourcegraph/cody/pull/1192
- Issue for Nathan's work is https://github.com/sourcegraph/sourcegraph/issues/56284, but this is quite out of date - https://github.com/sourcegraph/cody/pull/1192 has more context than the issue 😅 I'll try and get around to updating that tomorrow

**Comment by dominiccooney** [comment]
@nathan-downs , @bobheadxi thanks for those details. If the automated tests are enough it would be great for the test plan to show how to get that configuration and just run them.

CC @DanTup who is reordering some events to get a Windows integration test up, FYI. At some point we'll need the expected TelemetryV2 events in the right order.

**Comment by dominiccooney** [comment]
I posted on #1517 but it would be neat if you had `recordEvent` return an object which then does the classic `telemetry.log`. It will give you more leverage that the logs actually line up, and make it easier to delete all the old logging later.

**Comment by DanTup** [comment]
> CC @DanTup who is reordering some events to get a Windows integration test up, FYI. At some point we'll need the expected TelemetryV2 events in the right order.

My changes are in https://github.com/sourcegraph/cody/pull/1535 which was mostly making the events more predictable (less unwanted requests during the test). After both of these land, I could update those tests to verify the `TelemetryV2` events too (assuming they're available to verify in tests).

**Comment by nathan-downs** [comment]
replaced by https://github.com/sourcegraph/cody/pull/2060

**Pull request closed without merging**